cloud/misc/virt_net.py

```diff
@@ -0,0 +1,557 @@
+        choices: [ "define", "create", "start", "stop", "destroy",
+                   "undefine", "get_xml", "list_nets", "facts",
+                   "info", "status"]
+        description:
+            - in addition to state management, various non-idempotent commands are available.
+              See examples.
+    autostart:
+        required: false
+        choices: ["yes", "no"]
+        description:
+            - Specify if a given storage pool should be started automatically on system boot.
+    uri:
+        required: false
+        default: "qemu:///system"
+        description:
+            - libvirt connection uri.
+    xml:
+        required: false
+        description:
+            - XML document used with the define command.
+requirements:
+    - "python >= 2.6"
+    - "python-libvirt"
+    - "python-lxml"
+'''
+
+EXAMPLES = '''
+# Define a new network
+- virt_net: command=define name=br_nat xml='{{ lookup("template", "network/bridge.xml.j2") }}'
+
+# Start a network
+- virt_net: command=create name=br_nat
+
+# List available networks
+- virt_net: command=list_nets
+
+# Get XML data of a specified network
+- virt_net: command=get_xml name=br_nat
+
+# Stop a network
+- virt_net: command=destroy name=br_nat
+
+# Undefine a network
+- virt_net: command=undefine name=br_nat
+
+# Gather facts about networks
+# Facts will be available as 'ansible_libvirt_networks'
+- virt_net: command=facts
+
+# Gather information about network managed by 'libvirt' remotely using uri
+- virt_net: command=info uri='{{ item }}'
+  with_items: libvirt_uris
+  register: networks
+
+# Ensure that a network is active (needs to be defined and built first)
+- virt_net: state=active name=br_nat
+
+# Ensure that a network is inactive
+- virt_net: state=inactive name=br_nat
+
+# Ensure that a given network will be started at boot
+- virt_net: autostart=yes name=br_nat
+
+# Disable autostart for a given network
+- virt_net: autostart=no name=br_nat
+'''
+
+VIRT_FAILED = 1
+VIRT_SUCCESS = 0
+VIRT_UNAVAILABLE=2
+
+import sys
+
+try:
+    import libvirt
+except ImportError:
+    HAS_VIRT = False
+else:
+    HAS_VIRT = True
+
+try:
+    from lxml import etree
+except ImportError:
+    HAS_XML = False
+else:
+    HAS_XML = True
+
+ALL_COMMANDS = []
+ENTRY_COMMANDS = ['create', 'status', 'start', 'stop',
+                  'undefine', 'destroy', 'get_xml', 'define']
+HOST_COMMANDS = [ 'list_nets', 'facts', 'info' ]
+ALL_COMMANDS.extend(ENTRY_COMMANDS)
+ALL_COMMANDS.extend(HOST_COMMANDS)
+
+ENTRY_STATE_ACTIVE_MAP = {
+    0 : "inactive",
+    1 : "active"
+}
+
+ENTRY_STATE_AUTOSTART_MAP = {
+    0 : "no",
+    1 : "yes"
+}
+
+ENTRY_STATE_PERSISTENT_MAP = {
+    0 : "no",
+    1 : "yes"
+}
+
+class EntryNotFound(Exception):
+    pass
+
+
+class LibvirtConnection(object):
+
+    def __init__(self, uri, module):
+
+        self.module = module
+
+        cmd = "uname -r"
+        rc, stdout, stderr = self.module.run_command(cmd)
+
+        if "xen" in stdout:
```

Comment:
Actually, I took that code from `virt` module, which is kinda old, and I don't use Xen so hadn't had a chance to try it. Since Xen [has a valid URI](https://libvirt.org/drvxen.html) I could just drop above code and go stright to connecting to a specified URI without checking for Xen in the first place, not sure why that was/is there. I will do modifications with other changes. Thanks for the heads up.